src/geofabrics/bathymetry_estimation.py

```diff
@@ -0,0 +1,1576 @@
+                = self._rolling_mean_with_padding(cross_sections['min_z_centre_unimodal'],
+                                                  smoothing_samples)
+
+            # Smoothed slope
+            cross_sections[f'slope_mean_{label}'] = self._rolling_mean_with_padding(cross_sections['slope'],
+                                                                                    smoothing_samples)
+
+    def _smooth_widths_and_thresholds(self, cross_sections: geopandas.GeoDataFrame):
+        """ Record the valid and reolling mean of the calculated thresholds and widths
+
+        Parameters
+        ----------
+
+        cross_sections
+            The elevations and calculated widths and thresholds for each sampled cross section
+        """
+
+        invalid_mask = cross_sections['valid'] == False
+
+        # Tidy up widths - pull out the valid widths
+        cross_sections['valid_widths'] = cross_sections['widths']
+        cross_sections.loc[invalid_mask, 'valid_widths'] = numpy.nan
+        widths_no_nan = cross_sections['valid_widths'].interpolate('index', limit_direction='both')
+
+        # Flat widths
+        cross_sections['valid_flat_widths'] = cross_sections['flat_widths']
+        cross_sections.loc[invalid_mask, 'valid_flat_widths'] = numpy.nan
+        flat_widths_no_nan = cross_sections['valid_flat_widths'].interpolate('index', limit_direction='both')
+
+        # Tidy up thresholds - pull out the valid thresholds
+        cross_sections['valid_threhold'] = cross_sections['threshold']
+        cross_sections.loc[invalid_mask, 'valid_threhold'] = numpy.nan
+        thresholds_no_nan = cross_sections['valid_threhold'].interpolate('index', limit_direction='both')
+
+        # Cycle through and caluclate the rolling mean
+        for smoothing_distance in [150, 200, 250, 2000, 3000]:
+            # ensure odd number of samples so array length preserved
+            smoothing_samples = int(numpy.ceil(smoothing_distance / self.cross_section_spacing))
+            smoothing_samples = int(smoothing_samples / 2) * 2 + 1
+            label = f"{smoothing_distance/1000}km"
+
+            # Apply the rolling mean to each
+            cross_sections[f'widths_mean_{label}'] = self._rolling_mean_with_padding(widths_no_nan, smoothing_samples)
+            cross_sections[f'flat_widths_mean_{label}'] = self._rolling_mean_with_padding(flat_widths_no_nan, smoothing_samples)
+            cross_sections[f'thresholds_mean_{label}'] = self._rolling_mean_with_padding(thresholds_no_nan,
+                                                                                         smoothing_samples)
+
+            '''cross_sections[f'widths_Savgol_{label}'] = scipy.signal.savgol_filter(
+                cross_sections['widths'].interpolate('index', limit_direction='both'),
+                smoothing_samples,  # Ensure odd. number of samples included
+                3)  # Polynomial order'''
```

Comment:
Remove unused code or leave a comment on why it should be kept

Reply:
Fair point. I'll remove as not longer used and not adding anything in this case!